client/app/pages/dashboards/dashboard.js

```diff
@@ -137,8 +137,14 @@ function DashboardCtrl(
     this.extractGlobalParameters();
   });
 
-  const collectFilters = (dashboard, forceRefresh) => {
-    const queryResultPromises = _.compact(this.dashboard.widgets.map((widget) => {
+  const collectFilters = (dashboard, forceRefresh, pendingParameters = []) => {
```

Comment:
In here the parameters have already been updated, so perhaps `updatedParameters` instead?